bindings/rust/src/lib.rs

```diff
@@ -42,7 +42,20 @@ use std::ptr::NonNull;
 
 /// Parse and validate the input according to WebAssembly 1.0 rules. Returns true if the supplied input is valid.
 pub fn validate<T: AsRef<[u8]>>(input: T) -> bool {
-    unsafe { sys::fizzy_validate(input.as_ref().as_ptr(), input.as_ref().len()) }
+    let err = Box::new(sys::FizzyError {
+        code: 0,
+        message: [0i8; 256],
+    });
+    let err = Box::into_raw(err); // drop ownership
```

Comment:
It is possible to merge these two lines, but ideally if we there would be a way to retrieve the pointer without drop/claim ownership that would be nicer. Obviously this can be pulled out into a helper too.

Reply:
Pulled into a helper.